include/deal.II/fe/fe_nedelec.h

```diff
@@ -644,16 +644,261 @@ class FE_Nedelec : public FE_PolyTensor<dim>
+    {{0, 4, 8, 1, 5, 9, 2, 6, 10, 3, 7, 11}, // 3
+     {1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0},
+     {0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0}},
+    {{0, 4, 8, 1, 5, 9, 2, 6, 10, 3, 7, 11}, // 4
+     {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
+     {1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 5
+     {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
+     {1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 6
+     {1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0},
+     {0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0}},
+    {{0, 4, 8, 1, 5, 9, 2, 6, 10, 3, 7, 11}, // 7
+     {0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0},
+     {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0}}};
+
+  const std::vector<std::vector<std::vector<int>>> swap_table_4 = {
+    // Swap sign_X and sign_Y rows if k=4. Why?...
+    // 0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18
+    // 19
+    {{0,  5,  10, 15, 1,  6,  11, 16, 2,  7,
+      12, 17, 3,  8,  13, 18, 4,  9,  14, 19}, // 0
+     {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+     {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 1
+     {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+     {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 2
+     {1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1},
+     {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1}},
+    {{0,  5,  10, 15, 1,  6,  11, 16, 2,  7,
+      12, 17, 3,  8,  13, 18, 4,  9,  14, 19}, // 3
+     {0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1},
+     {1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1}},
+    {{0,  5,  10, 15, 1,  6,  11, 16, 2,  7,
+      12, 17, 3,  8,  13, 18, 4,  9,  14, 19}, // 4
+     {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
+     {1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 5
+     {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0},
+     {1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 0}},
+    {{-1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1}, // 6
+     {0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1},
+     {1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1}},
+    {{0,  5,  10, 15, 1,  6,  11, 16, 2,  7,
+      12, 17, 3,  8,  13, 18, 4,  9,  14, 19}, // 7
+     {1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1},
+     {0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1}}};
+
+  // Finally, the swap tables are organized into a vector.
+
+  const std::vector<std::vector<std::vector<std::vector<int>>>>
```

Comment:
There is no reason to store these inside the class (and for each instance of the element!). Please create a static variable in the .cc file instead. Additionally, you don't need the dynamic features of the ``std::vector``, can you use std::array or C arrays instead?

Reply:
Ok. I will try to do that. I will “git push” the new version when ready.

Reply:
Awesome, thank you! Let us know how we can help.